### PR TITLE
README.md for Extension directory

### PR DESCRIPTION
**Pull request by AFFogarty**
This small PR adds a `README.md` to the Extension directory.  For now, this file only contains a list of available extensions.  Eventually, it will be expanded to include instructions for how to develop a third-party extension.

I chose to make this a `README.md` and not a doc file in the `docs` directory because I think it is nice to have it immediately available when you look at the `Extensions` directory.  However, if folks feel strongly about moving it to the `docs` folder, I don't mind.

image

### CODE REVIEW

File: src/csharp/Microsoft.Spark.sln [hunk 1]
```diff
@@ -28,6 +28,9 @@ EndProject
 Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Microsoft.Spark.E2ETest.ExternalLibrary", "Microsoft.Spark.E2ETest.ExternalLibrary\Microsoft.Spark.E2ETest.ExternalLibrary.csproj", "{920A2CC8-4075-41D1-BC30-C496430BD9E4}"
 EndProject
 Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Extensions", "Extensions", "{71A19F75-8279-40AB-BEA0-7D4B153FC416}"
+	ProjectSection(SolutionItems) = preProject
```
Comment (imback82) [inline src/csharp/Microsoft.Spark.sln:31]: we don't seem to add other md files?

File: src/csharp/Extensions/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# .Net for Spark Extensions
+
+## Table of Contents
+- [NuGet Packages](#nuget-packages)
+
+## NuGet Packages
+
+The following .Net for Spark extensions are available as NuGet packages:
```
Comment (imback82) [inline src/csharp/Extensions/README.md:8]: `.NET for Apache Spark`

**Comment by imback82** [comment]
> I chose to make this a `README.md` and not a doc file in the `docs` directory because I think it is nice to have it immediately available when you look at the `Extensions` directory. However, if folks feel strongly about moving it to the `docs` folder, I don't mind.

This sounds fine to me.

File: src/csharp/Extensions/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# .Net for Spark Extensions
+
+## Table of Contents
```
Comment (suhsteve) [inline src/csharp/Extensions/README.md:3]: The ReadMe is very short.  Do we need the table of contents ?

Comment (AFFogarty) [reply src/csharp/Extensions/README.md:3]: I added it for future proofing.

File: src/csharp/Microsoft.Spark.sln (hunk 1 above)
Comment (AFFogarty) [reply src/csharp/Microsoft.Spark.sln]: Removed from `Microsoft.Spark.sln`.

File: src/csharp/Extensions/README.md (hunk 2 above)
Comment (AFFogarty) [reply src/csharp/Extensions/README.md]: Fixed.

**Pull request merged**